### ISSUE TITLE
Why does SqlClient for .Net Core not allow an authentication method 'Active Directory Interactive'?

### ISSUE THREAD

**Issue by acple** [dotnet/SqlClient#374]
I have been investigating what a way to use AzureAD Interactive authentication for Azure SQL Database on .Net Core runtime.
I created very cheap interactive authentication method for netcoreapp using MSAL.NET, realizes to use interactive authentication with .Net Core console application.
But the SqlConnection doesn't allow an `authentication` parameter with value `Active Directory Interactive` it throws a System.ArgumentException: Invalid value for key 'authentication'.

```csharp
// my custom provider
var provider = new LocalhostInteractiveSqlAuthenticationProvider(parameter.AzureClientId);
SqlAuthenticationProvider.SetProvider(SqlAuthenticationMethod.ActiveDirectoryInteractive, provider);

using var connection = new SqlConnection(parameter.ConnectionString); // ConnectionString: "...;Authentication=Active Directory Interactive;..."

// ...
```

```text
Unhandled exception. System.ArgumentException: Invalid value for key 'authentication'.
   at Microsoft.Data.Common.DbConnectionStringBuilderUtil.ConvertToAuthenticationType(String keyword, Object value)
   at Microsoft.Data.SqlClient.SqlConnectionString.ConvertValueToAuthenticationType()
   at Microsoft.Data.SqlClient.SqlConnectionString..ctor(String connectionString)
   at Microsoft.Data.SqlClient.SqlConnectionFactory.CreateConnectionOptions(String connectionString, DbConnectionOptions previous)       
   at Microsoft.Data.ProviderBase.DbConnectionFactory.GetConnectionPoolGroup(DbConnectionPoolKey key, DbConnectionPoolGroupOptions poolOptions, DbConnectionOptions& userConnectionOptions)
   at Microsoft.Data.SqlClient.SqlConnection.ConnectionString_Set(DbConnectionPoolKey key)
   at Microsoft.Data.SqlClient.SqlConnection.set_ConnectionString(String value)
   at Microsoft.Data.SqlClient.SqlConnection..ctor(String connectionString)
   at InteractiveAuthenticationTest.Program.Run(Parameter parameter) in C:\temp\InteractiveAuthenticationTest\Program.cs:line 41
   at InteractiveAuthenticationTest.Program.Main(String[] args) in C:\temp\InteractiveAuthenticationTest\Program.cs:line 33
   at InteractiveAuthenticationTest.Program.<Main>(String[] args)
```

I edited [TryConvertToAuthenticationType method](https://github.com/dotnet/SqlClient/blob/v1.1.0/src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/Common/DbConnectionStringCommon.cs#L122) to allow SqlAuthenticationMethod.ActiveDirectoryInteractive, then authentication works fine!

Now I know an interactive authentication can be work on .Net Core runtime, but SqlClient blocked to set interactive authentication. Why?
Or is there another way to use interactive authentication with SqlClient?
I want to use this over the EntityFrameworkCore, eventually.

### Information:

[The original authentication provider implementation and short testing code](https://github.com/acple/InteractiveAuthenticationTest/blob/master/Program.cs)

#### Env:
```
.NET Core SDK (reflecting any global.json):
 Version:   3.1.100
 Commit:    cd82f021f4

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.18363
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\3.1.100\

Host (useful for support):
  Version: 3.1.0
  Commit:  65f04fb6db
```

**Comment by cheenamalhotra** [comment]
Hi @acple 

We tried to support AAD Interactive for .NET Core when we implemented it for .NET Framework, but we were facing problems for cross platform support. Although it works in some scenarios, there are still some cases where MSAL.NET doesn't work as expected.

It is in our backlog and things to do, which will be picked up when it's re-prioritized.

**Comment by Wraith2** [comment]
When would you use interactive authentication to a database access other than through a tool like management studio?

**Comment by acple** [comment]
I see, is there no way to use interactive authentication with SqlClient now?

@Wraith2
Originally we want to use LINQPad6 with EF Core as a database acccess tool. It worked with older version that is on .Net Framework.

**Comment by David-Engel** [comment]
@acple You can acquire the AAD token in the application yourself and pass it in to SqlClient via the SqlConnection.AccessToken property.

**Comment by cheenamalhotra** [comment]
Hi @acple 

This has been addressed with [v2.0.0](https://github.com/dotnet/SqlClient/releases/tag/v2.0.0), closing issue as fixed.
